### PR TITLE
SEMS input on measures

### PR DESCRIPTION
**Pull request by benadida**
Still need to get the options for the measures to be included, because those "candidate IDs" need to carry through the election.json

### CODE REVIEW

File: sample_files/10_8-26-2020-expected-election.json [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+      "candidates": []
```
[the remark this answers is not in the document]
Comment (benadida) [reply sample_files/10_8-26-2020-expected-election.json:300]: good catch, and we need to update the data model anyways too 👍

File: converter/SEMSinput.py [hunk 2]
```diff
@@ -156,46 +159,140 @@ def process_row(row):
...
+    } if r[2] == "0" else {
+        "id": r[0],
+        "section": r[8],
+        "districtId": r[3],
+        "type": "yesno",
+        "title": cleanup_text(r[6]).split("\n")[0] + "\n" + r[1],
+        "description": "\n".join(cleanup_text(r[6]).split("\n")[1:])
+    } for r in c.execute(sql).fetchall() if r[1] not in either_neither_labels]
+
+    # either-or-contests
```
Comment (eventualbuddha) [inline converter/SEMSinput.py:186]:
```suggestion
    # ms-either-neither contests
```

File: converter/SEMSinput.py [hunk 3]
```diff
@@ -156,46 +159,140 @@ def process_row(row):
...
+    either_neither_contest_ids = []
+    for label in either_neither_labels:
+        sql = "select contest_id, contests.district_id, contest_text, districts.label, contests.label from contests, districts where contests.district_id = districts.district_id and contests.label = ? order by contest_id"
+        subcontests = c.execute(sql, [label]).fetchall()
+        text = cleanup_text(subcontests[0][2]).split("\n")
+        either_neither_contest_ids.append(subcontests[0][0])
+        either_neither_contest_ids.append(subcontests[1][0])
+        new_contest = {
+            "id": subcontests[0][0] + "-either-neither",
```
Comment (eventualbuddha) [inline converter/SEMSinput.py:195]: Should we include both ids? No strong preference here.

File: converter/SEMSinput.py [hunk 4]
```diff
@@ -156,46 +159,140 @@ def process_row(row):
...
+            "section": subcontests[0][3],
+            "districtId": subcontests[0][1],
+            "type": "ms-either-neither",
+            "title": text[0],
+            "eitherNeitherContestId": subcontests[0][0],
+            "pickOneContestId": subcontests[1][0],
+            "description": f"<b>{subcontests[0][4]}</b>\n\n" + "\n".join(text[1:])
```
Comment (eventualbuddha) [inline converter/SEMSinput.py:202]: These indexes are confusing.

File: converter/SEMSinput.py [hunk 5]
```diff
@@ -156,46 +159,140 @@ def process_row(row):
...
+            options = [{
+                "id": o[0],
+                "label": cleanup_text(o[2]).split("\n")[1]
+            } for o in c.execute(sql, [contest['id']]).fetchall()]
+
+            contest['yesOption'] = options[0]
+            contest['noOption'] = options[1]
+
+        if contest['type'] == 'ms-either-neither':
+            # in either or, we have two contests and each has a yes and a no, in that option order
```
Comment (eventualbuddha) [inline converter/SEMSinput.py:259]:
```suggestion
            # in either-neither, we have two contests and each has a yes and a no, in that option order
```

File: converter/SEMSinput.py (hunk 3 above)
Comment (benadida) [reply converter/SEMSinput.py:195]: probably a good safe thing to do!

File: converter/SEMSinput.py (hunk 4 above)
Comment (benadida) [reply converter/SEMSinput.py]: indeed, fixed to use dictionaries instead of arrays.

**Pull request merged**